### DISCUSSION TITLE
Toggling lights (on/off)

### DISCUSSION THREAD

**Discussion by mikevsky** [mitsuba-renderer/mitsuba3#402] [Feature requests]
Hey Mitsuba team!

I am trying to reproduce a setup, where an object is captured under different lighting conditions modelled with a set of light sources toggled on/off.

For example, I have 50 lights, 3 of them are "switched on", other 47 have zero radiance. I get the following look:
<img width="258" alt="image" src="https://user-images.githubusercontent.com/33276701/203308281-0f91fa4d-07b8-4871-bd53-82faf5dc2950.png">

If I remove the 47 "switched off" lights, I get far less grained pictures, e.g.:
<img width="408" alt="image" src="https://user-images.githubusercontent.com/33276701/203308389-86cddce5-a49f-4e29-b371-ae13e49416aa.png">

That would be amazing to have a feature to explicitly toggle the light off or just not account for it if it has zero radiance.

**Comment by Speierers** [comment]
Hi @waytobehigh,

The noise you see here in the first image comes from the emitter sampling routine that doesn't know which light is switched On/Off. It simply picks a light at random to build a light path. 

In order to achieve this you could implement your own integrator that would be aware of the state of the lights and only select lights that are ON when sampling an emitter.

Another (less efficient) option would be to reconstruct the scene only with the lights that are switched ON. But this will have a large overhead.

One final suggestion: create a new `Emitter` plugin (e.g. directly in Python) that represents all the lights, and internally only samples the ones that are switched ON. For instance, such a plugin would take a set of nested light plugins as properties.

**Reply by mikevsky** [reply]
Will try that, thank you!
